server/src/main/java/com/thoughtworks/go/config/PartialConfigService.java

```diff
@@ -69,21 +70,40 @@ public void onSuccessPartialConfig(ConfigRepoConfig repoConfig, PartialConfig in
 
         if (this.configWatchList.hasConfigRepoWithFingerprint(fingerprint)) {
             if (shouldMergePartial(incoming, fingerprint, repoConfig)) {
-                // validate rules before attempting updateConfig() so that
-                // rule violations will be considered before accepting a merge;
-                // updateConfig() only considers structural validity.
-                final boolean violatesRules = hasRuleViolations(incoming);
-
+                // mark the fingerprint as last known
                 cachedGoPartials.cacheAsLastKnown(fingerprint, incoming);
 
-                if (updateConfig(incoming, fingerprint, repoConfig)) {
+                //validate rules
+                hasRuleViolations(incoming);
```

Comment:
@GaneshSPatil do we need to explicitly check for rules violation here? Wouldn't config update apply rules?